rs/nns/sns-wasm/src/sns_wasm.rs

```diff
@@ -1723,6 +1743,56 @@ where
                 )
             })
     }
+
+    // TODO[NNS1-2997]: Remove this function once the migration is completed.
+    pub async fn populate_wasm_metadata(
+        thread_safe_sns: &'static LocalKey<RefCell<SnsWasmCanister<M>>>,
+    ) {
+        // 2. Compute metadata for all marked WASMs.
+        let all_wasm_hashes = thread_safe_sns.with(|sns_canister| {
+            sns_canister
+                .borrow()
+                .wasm_indexes
+                .keys()
+                .cloned()
+                .collect::<Vec<_>>()
+        });
+        for hash in all_wasm_hashes {
+            ic_cdk_timers::set_timer(std::time::Duration::from_millis(0), move || {
+                let future = async move {
+                    thread_safe_sns.with(|sns_canister| {
+                        if let Some(index) = sns_canister.borrow().wasm_indexes.get(&hash) {
+                            if index.metadata == vec![] {
+                                let metadata = sns_canister
+                                    .borrow()
+                                    .get_wasm_metadata_impl(GetWasmMetadataRequestPb {
```

Comment:
Is there a reason to do this as a request?  Instead of using the Self:: read_wasm_metadata_or_err implementation, and then allowing get_wasm_metadata_impl to read from the cache?

Reply:
I tried both, this resulted in fewer lines of code.